flux.tf

```diff
@@ -0,0 +1,87 @@
+resource "kubernetes_namespace" "flux" {
+  count      = var.flux_enabled ? 1 : 0
+
+  metadata {
+    name = "flux"
+  }
+}
+
+resource "kubernetes_secret" "flux_secret" {
+  count      = var.flux_enabled ? 1 : 0
+  
+  metadata {
+    name      = "flux-secret"
+    namespace = kubernetes_namespace.flux.0.metadata.0.name
+  }
+
+  data = {
+    "identity" = var.flux_private_key_pem
+  }
+}
+
+resource "helm_release" "helm_operator" {
+  count      = var.flux_enabled ? 1 : 0
```

Comment:
Have you tested this? I wanted to make the same thing in the multi-region, but the problem is that if you use count and specify the namespace like this `kubernetes_secret.flux_secret.0.metadata.0.name` you will have an error.

Reply:
Everything is tested and works as expected.